### ISSUE TITLE
Burn Book: Introduction to Rust

### ISSUE THREAD

**Issue by louisfd** [tracel-ai/burn#1155]
Add an introduction to rust section
For people who don't know Rust, following the guide should be enough to get running with Burn
Refer to the most relevant sections of the Rust book
